enumerable.rb

```diff
@@ -0,0 +1,119 @@
+# Enumerable method
+module Enumerable
+  def my_each
+    arr = self
+    if arr.is_a?(Array)
+      (0..(arr.length - 1)).each do |i|
+        yield(arr[i])
+      end
+    end
+    arr
+  end
+
+  def my_each_with_index
+    arr = self
+    (0..(arr.length - 1)).each do |i|
+      yield(arr[i], i)
+    end
+    arr
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result.push(item) if yield(item)
+    end
+    puts result.to_s
+    result
+  end
+
+  def my_all?
+    my_each do |num|
+      if yield(num) == false
+        puts false
+        return false
+      end
+    end
+    puts true
+    true
+  end
+
+  def my_any?
+    my_each do |num|
+      if yield(num) == true
+        puts true
+        return true
+      end
+    end
+    puts false
+    false
+  end
+
+  def my_none?
+    my_each do |num|
+      if yield(num) == true
+        puts false
+        return false
+      end
+    end
+    puts true
+    true
+  end
+
+  def my_count
+    count = 0
+    my_each do |_acc|
+      count += 1
+    end
+    puts count
+    count
+  end
+
+  def my_map
+    array = []
+    my_each do |item|
+      array.push(yield(item))
+    end
+    puts array.to_s
+    array
+  end
```

Comment:
- #my_map returns a new array with the results of running a given block once for every element in enum::range
-#my_map returns an Enumerator if no block is given
- #my_map executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.